plugins/plugin-core-metric-middleware-statistics/lib/statistics/Statistic.ts

```diff
@@ -57,43 +56,76 @@ export abstract class Statistic {
     callback: (newPropertyName: string, distribution: Distribution) => void
   ) {
     for (const [distributionName, distribution] of distributions) {
-      callback(this.getPropertyName(distributionName), distribution);
+      callback(`${this.name}-${distributionName}`, distribution);
     }
   }
 
   loopThroughSeries(
     series: SeriesMap<number>,
     callback: (
       newPropertyName: string,
-      seriesByType: Map<SeriesUnit, Series<number>>
+      seriesName: MetricName,
+      seriesUnit: SeriesUnit,
+      series: Series<number>
     ) => void
   ) {
-    for (const [seriesName, seriesByType] of series) {
-      callback(this.getPropertyName(seriesName), seriesByType);
+    for (const [seriesName, seriesByUnit] of series) {
+      for (const [seriesUnit, series_] of seriesByUnit) {
+        callback(
+          `${this.name}-${seriesName}-${seriesUnit}`,
+          seriesName,
+          seriesUnit,
+          series_
+        );
+      }
     }
   }
 
   loopThroughSeriesDistribution(
     seriesDistributions: SeriesMap<Distribution>,
     callback: (
       newPropertyName: string,
-      seriesByType: Map<SeriesUnit, Series<Distribution>>
+      seriesDistributionName: MetricName,
+      seriesUnit: SeriesUnit,
+      series: Series<Distribution>
     ) => void
   ) {
-    for (const [distributionName, distribution] of seriesDistributions) {
-      callback(this.getPropertyName(distributionName), distribution);
+    for (const [
+      seriesDistributionName,
+      seriesDistributionByUnit,
+    ] of seriesDistributions) {
+      for (const [seriesUnit, series] of seriesDistributionByUnit) {
+        callback(
+          `${this.name}-${seriesDistributionName}-${seriesUnit}`,
+          seriesDistributionName,
+          seriesUnit,
+          series
+        );
+      }
     }
   }
 
   loopThroughSeriesMeasurement(
     seriesMeasurements: SeriesMap<PropertiesMap<number>>,
     callback: (
       newPropertyName: string,
-      seriesByType: Map<SeriesUnit, Series<PropertiesMap<number>>>
+      seriesDistributionName: MetricName,
+      seriesUnit: SeriesUnit,
+      series: Series<PropertiesMap<number>>
     ) => void
   ) {
-    for (const [name, series] of seriesMeasurements) {
-      callback(this.getPropertyName(name), series);
+    for (const [
+      seriesMeasurementName,
+      seriesMeasurementsByUnit,
+    ] of seriesMeasurements) {
+      for (const [seriesUnit, series] of seriesMeasurementsByUnit) {
+        callback(
+          `${this.name}-${seriesMeasurementName}-${seriesUnit}`,
+          seriesMeasurementName,
+          seriesUnit,
+          series
+        );
+      }
     }
   }
 
```

Comment:
Since you removed the usage of the function below, maybe we can remove it.

Reply:
I removed it